.github/scripts/yarn-dedupe.sh

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+
+set -ex
+
+if git diff --exit-code yarn.lock; then
+  echo "✅ yarn.lock is already properly deduplicated"
+else
+  echo "📝 yarn.lock was modified by dedupe command"
+
+  PR_AUTHOR="${PR_AUTHOR:-}"
+  PR_USER_TYPE="${PR_USER_TYPE:-}"
+  IS_BOT=false
+
+  if [[ "$PR_USER_TYPE" == "Bot" ]] || [[ "$PR_AUTHOR" == *"bot"* ]] || [[ "$PR_AUTHOR" == "renovate"* ]] || [[ "$PR_AUTHOR" == "github-actions"* ]]; then
```

Comment:
I think you wanted to write dependabot, not renovate? :)

Reply:
No it was actually intentional. It was based of a question I asked AI on how to match bots in general in a GitHub Actions workflow. The main way to detect it is if `PR_USER_TYPE` is set to `Bot`, which many bots use these days. If that's not used by the bot correctly, fall back to check if it has the word `bot` in its name (which Dependabot does), if not, then check if it starts with either `renovate` or `github-actions`. But if you want me to simplify it I can. E.g. just check for `PR_USER_TYPE == "Bot"`, which I believe Dependabot is correctly is identifying itself as

Reply:
Fixed 👍